### DISCUSSION TITLE
A Question on link splitting

### DISCUSSION THREAD

**Discussion by Razavi1999** [pyg-team/pytorch_geometric#4051] [Q&A]
I was wondering why you are returning upper triangular portion of the edges in at this [link](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/utils/train_test_split_edges.py#:~:text=%23%20Return%20upper%20triangular%20portion).
If you could tell me the theory of why this is happening in here, I would really appreciate that.

**Comment by rusty1s** [comment]
`train_test_split_edges` assumes undirected graphs, and so we solely split on the upper triangular portion of the graph so that reverse edges do not land in different splits.
